src/Core/Domain/Product/Stock/QueryResult/RangeStockMovementHistory.php

```diff
@@ -0,0 +1,204 @@
+     */
+    protected $toDate = null;
+
+    public function __construct(
+        int $deltaQuantity,
+        DateTimeInterface $fromDate,
+        DateTimeInterface $toDate
+    ) {
+        $this->deltaQuantity = $deltaQuantity;
+        $this->fromDate = $fromDate;
+        $this->toDate = $toDate;
+    }
+
+    public function getStockMovementIds(): array
+    {
+        return $this->stockMovementIds;
+    }
+
+    public function setStockMovementIds(int ...$stockMovementIds): self
+    {
+        $this->stockMovementIds = $stockMovementIds;
+
+        return $this;
+    }
+
+    public function getStockIds(): array
+    {
+        return $this->stockIds;
+    }
+
+    public function setStockIds(int ...$stockIds): self
+    {
+        $this->stockIds = $stockIds;
+
+        return $this;
+    }
+
+    public function getStockMovementReasonIds(): array
+    {
+        return $this->stockMovementReasonIds;
+    }
+
+    public function setStockMovementReasonIds(int ...$stockMovementReasonIds): self
+    {
+        $this->stockMovementReasonIds = $stockMovementReasonIds;
+
+        return $this;
+    }
+
+    public function getOrderIds(): array
+    {
+        return $this->orderIds;
+    }
+
+    public function setOrderIds(int ...$orderIds): self
+    {
+        $this->orderIds = $orderIds;
+
+        return $this;
+    }
+
+    public function getEmployeeIds(): array
+    {
+        return $this->employeeIds;
+    }
+
+    public function setEmployeeIds(int ...$employeeIds): self
+    {
+        $this->employeeIds = $employeeIds;
+
+        return $this;
+    }
+
+    public function getEmployeeName(?TranslatorInterface $translator = null): ?string
+    {
+        return $this->employeeName;
+    }
+
+    public function setEmployeeName(?string $employeeName): self
+    {
+        $this->employeeName = $employeeName;
+
+        return $this;
+    }
+
+    public function getDeltaQuantity(): int
+    {
+        return $this->deltaQuantity;
+    }
+
+    public function getFromDate(): DateTimeInterface
+    {
+        return $this->fromDate;
+    }
+
+    public function getToDate(): DateTimeInterface
+    {
+        return $this->toDate;
+    }
+
+    public function getDates(): array
+    {
+        return [
+            'from' => $this->getFromDate(),
+            'to' => $this->getToDate(),
+        ];
+    }
+
+    public function getDateRange(?TranslatorInterface $translator = null): string
+    {
+        $dateStrings = [];
+
+        foreach (array_values($this->getDates()) as $index => $date) {
+            $dateStrings['date' . ($index + 1)] = $date->format(DateTime::DEFAULT_DATETIME_FORMAT);
+        }
+        if ($translator instanceof TranslatorInterface) {
+            return $translator->trans(
+                'From %date1% to %date2%',
+                $dateStrings,
+                'Admin.Orderscustomers.Feature'
+            );
+        }
+
+        return vsprintf('%s - %s', array_values($dateStrings));
+    }
```

Comment:
```suggestion
    public function getFromDate(?bool $formated = false): DateTimeInterface
    {
        $date = $this->fromDate;
        return $formated ? $date->format(DateTime::DEFAULT_DATETIME_FORMAT) : $date;
    }

    public function getToDate(?bool $formated = false): DateTimeInterface
    {
        $date = $this->toDate;
        return $formated ? $date->format(DateTime::DEFAULT_DATETIME_FORMAT) : $date;
    }

    public function getDates(?bool $formated = false): array
    {
        return [
            'from' => $this->getFromDate($formated),
            'to' => $this->getToDate($formated),
        ];
    }

    public function getDateRange(?TranslatorInterface $translator = null): string
    {
        if (null != $translator) {
            return $translator->trans(
                'From %date_from% to %date_to%',
                [
                    '%date_from%' =>  $this->getFromDate(true),
                    %date_to%' => $this->getToDate(true),
                ],
                'Admin.Orderscustomers.Feature'
            );
        }

        return $this->getFromDate(true). ' - ' . $this->getToDate(true);
    }
```

Reply:
@PululuK your suggestion is invalid. Return type has to be unique: `DateTimeInterface`, not `DateTimeInterface|string`.
In my humble opinion, it is bad practice to use an optional argument to select the method behavior.